### ISSUE TITLE
💅 svelte specific syntax isn't tracked leading to false positive noUnusedVariables lint error

### ISSUE THREAD

**Issue by machadinhos** [biomejs/biome#11215]
### Environment information

<details>

```bash

N.A.

```

</details>

### Rule name

noUnusedVariables

### Playground link

https://biomejs.dev/playground/?indentStyle=space&tab=formatter&pane=Diagnostics&code=PABzAGMAcgBpAHAAdAAgAGcAZQBuAGUAcgBpAGMAcwA9ACIAVAAgAGUAeAB0AGUAbgBkAHMAIABCAGwAbwBiACIAIABsAGEAbgBnAD0AIgB0AHMAIgA%2BAAoAIAAgAGkAbQBwAG8AcgB0ACAAdAB5AHAAZQAgAHsAIABCAGwAbwBiACAAfQAgAGYAcgBvAG0AIAAiAG4AbwBkAGUAOgBiAHUAZgBmAGUAcgAiADsACgAKACAAIABjAG8AbgBzAHQAIABjAG8AbABvAHIAIAA9ACAAIgAxADIAMwAiADsACgAgACAAYwBvAG4AcwB0ACAAbQB5AEMAbABhAHMAcwAgAD0AIAB0AHIAdQBlADsACgAgACAAYwBvAG4AcwB0ACAAdgBhAGwAdQBlADoAIABUACAAfAAgAHUAbgBkAGUAZgBpAG4AZQBkADsACgA8AC8AcwBjAHIAaQBwAHQAPgAKAAoAewBhAH0ACgAKADwAZABpAHYAIABiAGkAbgBkADoAYwBvAGwAbwByAD4AZgBvAG8APAAvAGQAaQB2AD4ACgA8AGQAaQB2ACAAYwBsAGEAcwBzADoAbQB5AEMAbABhAHMAcwA%2BAGIAYQByADwALwBkAGkAdgA%2BAAoA&language=svelte

### Expected result

4 errors/warnings appear:
```
Error parse: Const declarations must have an initialized value.
Warning lint/correctness/noUnusedImports: This import is unused.
Warning lint/correctness/noUnusedVariables: This variable color is unused.
Warning lint/correctness/noUnusedVariables: This variable myClass is unused.
```

The error and first warning are caused because biome doesn't track/recognize the `generics` attribute you can pass to the `script` tag.
You can define multiple generics by separating them with commas `generics="T, U, V"` like you would do for type params in regular typescript `<T, U, V>`, and they also allow `const` prefixing `generics="const T extends Record<string, number>"` same as in regular typescript.
[link to svelte docs](https://svelte.dev/docs/svelte/typescript#Generic-$props)

The 2 last warnings happen because biome doesn't track variables that are used with the svelte shorthand syntax, for class and style, for when the value you want to set has the same name as the variable.
[link to svelte docs for style](https://svelte.dev/docs/svelte/style)
[link to svelte docs for class](https://svelte.dev/docs/svelte/class#The-class:-directive)

None of these warnings/errors should happen.

(I'm not using the `generics` for a prop in the example, which is the intended use, but the issue remains even if you assign generics to props)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Lihu0** [comment]
try adding this to your config
```json
{
  "html": {
    "experimentalFullSupportEnabled": true
  }
}

```

**Comment by machadinhos** [comment]
If you look at the biome playground link I provided you will see that that option is enabled and the errors are still there.
